pkg/function/copy_volume_data.go

```diff
@@ -99,7 +99,9 @@ func copyVolumeDataPodFunc(cli kubernetes.Interface, tp param.TemplateParams, na
 			return nil, errors.Wrapf(err, "Failed to write credentials to Pod %s", pc.PodName())
 		}
 
-		defer remover.Remove(context.Background()) //nolint:errcheck
+		if remover != nil { // WriteCredsToPod might return nil when nothing was written
```

Comment:
Check my PR comment: https://github.com/kanisterio/kanister/pull/2072#issuecomment-1560867072

Reply:
Ok, I agree with NopRemover. I thought about this but decided that probably it's an overkill. But since this came not only to my mind, it probably makes sense.